Update npm "prepare" script, CONTRIBUTING & README docs

BeniCheni opened the pull request:
* Add `--hoist` flag to lerna publishing in npm "prepare" script.
* Update `CONTRIBUTING.md` to add content around local development workflow.
* Introduce "Highlight topics" to quick links to useful sections in both `CONTRIBUTING.md` & `README.md`.
* Update misc. markdown format:
  * up to 80 characters per line
  * bottom link variables
  * whitespace cleanup.
* For some reason, there is an test snapshot update for slider's index.js.

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -1,8 +1,8 @@
 If you'd like to contribute to the design system, we'd love to have your help.
-As with any open source project, we ask that you be nice, professional, and courteous towards others.
+As with any open source project, we ask that you be nice, professional, and
+courteous towards others.
```
Thanks for cleaning this up a bit. Might be worth enabling prettier for markdown, which would handle this automatically going forward

BeniCheni replied:
Nice suggestion. Created #428 to track the suggested topic of prettier for markdown.

jxnblk replied:
This is such a small change, I was thinking we could just get it in in this PR...

BeniCheni replied:
Turned out #429 has been opened to take care of #428.

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -11,12 +11,25 @@ Contributing doesn't necessarily mean commiting code, we also encourage you to:
 
+Highlight topics:
+
+* [Development](#development)
```
I think this might make more sense as a table of contents at the top of the file

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -25,21 +38,27 @@ cd design-system
 [Lerna]: https://lernajs.io
 
-### Install dependencies:
+### Install dependencies
```
👍

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -73,18 +93,21 @@ open http://localhost:8000/
 ### Static Docs Site
 
-Markdown and source code for the [static docs site](https://pricelinelabs.github.io/design-system/) are located in `docs/`.
+Markdown and source code for the [static docs site][static-docs-site] are
```
NB: when the link name is the same as the shortcut, it will automatically create a link – if you take this suggestion, it'll also need updated below

> The implicit link name shortcut allows you to omit the name of the link, in which case the link text itself is used as the name.

```suggestion
 Markdown and source code for the [docs site][] are
```

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -150,10 +180,10 @@ Generally follow these rules for creating a PR:
 [jest]: https://facebook.github.io/jest/
+[semver]: http://semver.org
 [snapshots]: https://facebook.github.io/jest/docs/en/snapshot-testing.html#content
+[static-docs-site]: https://pricelinelabs.github.io/design-system/
```
```suggestion
 [docs site]: https://pricelinelabs.github.io/design-system/
```

jxnblk commented on package.json:
```diff
@@ -2,7 +2,7 @@
   "scripts": {
     "precommit": "lint-staged",
-    "prepare": "lerna bootstrap",
+    "prepare": "lerna bootstrap --hoist",
```
I think the change here is why the snapshot changed

jxnblk commented on CONTRIBUTING.md:
```diff
@@ -11,12 +11,25 @@ Contributing doesn't necessarily mean commiting code, we also encourage you to:
+Highlight topics:
+
+* [Development](#development)
+* [Lerna](#lerna)
```
nit: I think we tend to use `-` for lists elsewhere in markdown

BeniCheni commented:
@jxnblk,

> ... make more sense as a table of contents at the top of the file
> ... tend to use - for lists elsewhere in markdown

Updated to have a list of bullets as TOC at top of `CONTRIBUTING.md` (using `-` as list syntax). The "Highlight topics" in `README.md` still highlights the helpful topics to link to the TOC.

> ... prepare runs automatically after install

Tested again locally, & confirmed that `prepare` is no needed as part of the `npm i` workflow. Updated the content in the "**Install dependencies**" section of `CONTRIBUTING.md`.

Please review again at convenience.

jxnblk commented on README.md:
```diff
@@ -20,9 +13,23 @@ npm install pcln-design-system
+For all contributors, please be sure to read the
+[Contributing](CONTRIBUTING.md) doc.
+
+Highlight topics:
```
With the TOC in the contributing doc, do we need this section? Seems like it might go out of sync over time

BeniCheni commented:
> With the TOC in the contributing doc, do we need this section? Seems like it might go out of sync over time

Good catch. Remove "Highlight topics" from `README.md`. 🙏 👍 @jxnblk

jxnblk commented:
Looks like the lock file or something went out of sync with the last commit??

BeniCheni commented:
Confirmed "_green_" CI and the clean code diff to land this PR cleanly. LGTM to land now to `next` branch now.
